### PR TITLE
user account model and api

### PR DESCRIPTION
**Pull request by Mxia09**
user account model and api created in recent TS formatting

### CODE REVIEW

File: server/src/authServer.ts [hunk 1]
```diff
@@ -30,6 +31,30 @@ app.get('/posts', authenticateToken, (req: Request, res: Response) => {
   res.json(posts.filter(post => post.username === user.name))
 })
 
+app.post('/register', (req: Request, res: Response) => {
+  const { username, password }: { username: string; password: string } = req.body
+
+  bcrypt
+    .hash(password, 10)
```
Comment (bkeys818) [inline server/src/authServer.ts:38]: Good job hashing the password! We need to salt it as well. Here's an [awesome video](https://www.youtube.com/watch?v=--tnZMuoK3E) explaining what a password hash, salt, and pepper each are.

This [StackOverflow post](https://stackoverflow.com/questions/43092071/how-should-i-store-salts-and-passwords-in-mongodb) shows a very basic example of how to use password salts. Notice how the password never leaves the backend.

**Review by bkeys818 — CHANGES_REQUESTED** [review]

**Comment by bkeys818** [comment]
Sorry for hitting you with so many notes. If you need any help, I actually don't have much experience with this and would love to learn with you.

File: server/src/api/accounts.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const { username, firstname, lastname, email, password } = req.body
+  const accountDoc = await Account.create({ username, firstname, lastname, email, password })
+  res.json(accountDoc)
+})
+
+export type UpdateRequestBody = Omit<Account, '_id' | 'account'>
+export type UpdateResponseBody = Account
+
+accountRouter.put('/:id', async (req, res) => {
+  const { title, amount, description, frequency, date, category } = req.body
```
[the remark this answers is not in the document]
Comment (Mxia09) [reply server/src/api/accounts.ts:26]: yikes used expenses you created during the ts transition as the reference. Fixing that ASAP

File: server/src/api/accounts.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+accountRouter.get('/:id', async (req, res) => {
+  const account = await Account.findOne({ _id: req.params.id }).exec()
+  res.json(account)
+})
+
+export type CreateRequestBody = Omit<Account, '_id'>
+export type CreateResponseBody = Account
+
+accountRouter.post('/', async (req, res) => {
```
[the remark this answers is not in the document]
Comment (Mxia09) [reply server/src/api/accounts.ts:16]: This is just a test endpoint for the JWT token. If a user has been able to login and and the server generates a token that is specific to them they will be able to view information that is specific to them with their token. So in practice if User 1 logs in and the token generated is tied to them they will be the only one who can view their information.

File: server/src/authServer.ts (hunk 1 above)
Comment (Mxia09) [reply server/src/authServer.ts:38]: awesome thanks for the suggestion!

File: server/src/authServer.ts [hunk 4]
```diff
@@ -30,6 +31,30 @@ app.get('/posts', authenticateToken, (req: Request, res: Response) => {
   res.json(posts.filter(post => post.username === user.name))
 })
 
+app.post('/register', (req: Request, res: Response) => {
```
[the remark this answers is not in the document]
Comment (Mxia09) [reply server/src/authServer.ts:34]: you got it! refactoring this now

File: server/src/api/accounts.ts
[the remark this answers is not in the document]
Comment (Mxia09) [reply server/src/api/accounts.ts:1]: definitively let me know when you are free to go on a call.

**Review by bkeys818 — APPROVED** [review]

**Pull request merged**